src/ssat/utils/ParseUtils.h

```diff
@@ -95,24 +96,28 @@ static int parseInt(B& in) {
 // parse float for SSAT solving
 template<class B>
 static double parseFloat(B& in) {
-    int     val = 0;
-    int     deg = 0;
-    double  flt;
     skipWhitespace(in);
+
+    // make sure the floating point value starts with "0."
     if ( *in != '0' ) fprintf( stderr , "PARSE ERROR! Unexpected char: %c\n", *in ) , exit(3);
     else {
        ++in;
        if ( *in != char(46) ) fprintf( stderr , "PARSE ERROR! Unexpected char: %c\n", *in ) , exit(3);
+        ++in;
     }
-    ++in;
+
+    // collect all digits
+    std::string sbuf = "0.";
+    int cnt = 0;
     while ( *in >= '0' && *in <= '9' ) {
-       val = val*10 + (*in - '0');
+       sbuf += *in;
        ++in;
-       ++deg;
+       ++cnt;
+    }
+    if (cnt > 15) {
+       fprintf( stderr , "PARSE WARNING! Precision of floating-point number is only up to 15th digit, got %d\n", cnt);
     }
-    flt = (double)val;
-    for ( int i = 0 ; i < deg ; ++i ) flt *= 0.1;
-    return flt; 
+    return std::stod(sbuf);
```

Comment:
I am not familiar with `std::stod`, but from the [manual page](http://www.cplusplus.com/reference/string/stod/), it should be able to parse floating-point numbers with both integral and decimal parts.

Reply:
Yes, that's right.